cogs/surveys.py

```diff
@@ -33,9 +38,8 @@ async def on_message(self, message: discord.Message) -> None:
         """
         with suppress(AttributeError):
             if (
-                message.channel.name != "surveys"
-                and "276969339901444096"  # Why is this still hardcoded?
-                # TODO: Add role to env
+                message.channel.id
+                != self.allow_survey_channel_id
                 not in [role.id for role in message.author.roles]
                 and not message.author.bot
```

Comment:
this is parsed as `(message.channel.id != self.allow_survey_channel_id) not in [role.id for role in message.author.roles]`
which will become `False not in [role.id for role in message.author.roles]`
You'll need to add back in the role ID like
```py
message.channel.id != self.allow_survey_channel_id
and special_role_id not in [role.id for role in message.author.roles]
and not message.author.bot
```

Reply:
The newest commit should fix this error, ironically the old system didn't work either since it was trying to compare a string to a list of ints